src/main/scala/URAlgorithm.scala

```diff
@@ -122,469 +139,555 @@ case class URAlgorithmParams(
   userBias: Option[Float] = None, // will cause the default search engine boost of 1.0
   itemBias: Option[Float] = None, // will cause the default search engine boost of 1.0
   returnSelf: Option[Boolean] = None, // query building logic defaults this to false
-  fields: Option[List[Field]] = None, //defaults to no fields
+  fields: Option[Seq[Field]] = None, //defaults to no fields
   // leave out for default or popular
-  backfillField: Option[BackfillField] = None,
-  // name of date property field for when the item is avalable
+  backfills: Option[Seq[BackfillParams]] = None,
+  // name of date property field for when the item is available
   availableDateName: Option[String] = None,
   // name of date property field for when an item is no longer available
   expireDateName: Option[String] = None,
   // used as the subject of a dateRange in queries, specifies the name of the item property
   dateName: Option[String] = None,
-  seed: Option[Long] = None) // seed is not used presently
-  extends Params //fixed default make it reproducible unless supplied
+  seed: Option[Long] = None // seed is not used presently
+) extends Params //fixed default make it reproducible unless supplied
 
 /** Creates cooccurrence, cross-cooccurrence and eventually content correlators with
-  * [[org.apache.mahout.math.cf.SimilarityAnalysis]] The analysis part of the recommender is
-  * done here but the algorithm can predict only when the coocurrence data is indexed in a
-  * search engine like Elasticsearch. This is done in URModel.save.
-  *
-  * @param ap taken from engine.json to describe limits and event types
-  */
+ *  [[org.apache.mahout.math.cf.SimilarityAnalysis]] The analysis part of the recommender is
+ *  done here but the algorithm can predict only when the coocurrence data is indexed in a
+ *  search engine like Elasticsearch. This is done in URModel.save.
+ *
+ *  @param ap taken from engine.json to describe limits and event types
+ */
 class URAlgorithm(val ap: URAlgorithmParams)
   extends P2LAlgorithm[PreparedData, NullModel, Query, PredictedResult] {
 
-  case class BoostableCorrelators(actionName: String, itemIDs: Seq[String], boost: Option[Float])
-  case class FilterCorrelators(actionName: String, itemIDs: Seq[String])
+  @transient lazy implicit val logger: Logger = Logger[this.type]
 
-  @transient lazy val logger = Logger[this.type]
+  case class BoostableCorrelators(actionName: String, itemIDs: Seq[ItemID], boost: Option[Float]) {
+    def toFilterCorrelators: FilterCorrelators = {
+      FilterCorrelators(actionName, itemIDs)
+    }
+  }
+  case class FilterCorrelators(actionName: String, itemIDs: Seq[ItemID])
+
+  val appName: String = ap.appName
+  val recsModel: String = ap.recsModel.getOrElse(defaultURAlgorithmParams.DefaultRecsModel)
+  val eventNames: Seq[String] = ap.eventNames
+
+  val userBias: Float = ap.userBias.getOrElse(1f)
+  val itemBias: Float = ap.itemBias.getOrElse(1f)
```

Comment:
I think It is better to *not* send a bias to Elasticsearch, in other words leave it blank, than to send a bias of 1.0.